packages/foundry/script/DeployHelpers.s.sol

```diff
@@ -15,20 +19,42 @@ contract ScaffoldETHDeploy is Script {
   string root;
   string path;
   Deployment[] public deployments;
+  uint256 constant ANVIL_LAST_PK =
+    0x2a871d0798f97d79848a013d4936a73bf4cc922c825d33c1cf7073dff6d409c6;
+  address constant ANVIL_LAST_ACCOUNT =
+    0xa0Ee7A142d267C1f36714E4a8F75612F20a79720;
+  uint256 constant ANVIL_BASE_BALANCE = 1000 ether;
+
+  function _startBroadcast() internal returns (address deployer) {
+    vm.startBroadcast();
+    (, deployer,) = vm.readCallers();
+
+    if (block.chainid == 31337 && deployer.balance) {
+      vm.stopBroadcast();
+
+      // ------------- FUND DEPLOYER ACCOUNT -------------
+      vm.startBroadcast(ANVIL_LAST_PK);
+      (bool success,) = deployer.call{ value: ANVIL_BASE_BALANCE / 2 }("");
+      if (!success) {
+        revert DeployerHasNoBalance();
+      } else {
+        try this.anvil_setBalance(ANVIL_LAST_ACCOUNT, ANVIL_BASE_BALANCE) {
+          emit AnvilSetBalance(ANVIL_LAST_ACCOUNT, ANVIL_BASE_BALANCE);
+        } catch {
+          revert FailedAnvilRequest();
+        }
```

Comment:
question: Since we are setting the balance here directly, can't we just directly set the balance of `deployer` here? 

Instead of first sending fund from anvil last account to deployer then resetting the balance of anvil last account?

Reply:
i tried doing this first and it seems that the deployer will get the balance only after the a transaction is run , i.e. if we do `yarn deploy` again. This is not ideal because we want to deploy on this transaction using the deployer account.